staging/src/k8s.io/apiserver/pkg/cel/library/cost.go

```diff
@@ -126,13 +126,32 @@ func (l *CostEstimator) EstimateCallCost(function, overloadId string, target *ch
 			sz := l.sizeEstimate(*target)
 			toReplaceSz := l.sizeEstimate(args[0])
 			replaceWithSz := l.sizeEstimate(args[1])
-			// smallest possible result: smallest input size composed of the largest possible substrings being replaced by smallest possible replacement
-			minSz := uint64(math.Ceil(float64(sz.Min)/float64(toReplaceSz.Max))) * replaceWithSz.Min
-			// largest possible result: largest input size composed of the smallest possible substrings being replaced by largest possible replacement
-			maxSz := uint64(math.Ceil(float64(sz.Max)/float64(toReplaceSz.Min))) * replaceWithSz.Max
+
+			var perReplaceWithSz, retainedSz checker.SizeEstimate
+			if toReplaceSz.Min == 0 {
+				// if the string being replaced is empty, replace surrounds all characters in the input string with the replacement.
+				perReplaceWithSz.Max = sz.Max + 1
+				// and the longest possible original string length is retained
+				retainedSz.Max = sz.Max
+			} else {
+				// largest input size composed of the smallest possible substrings being replaced by largest possible replacement
+				// as many times as possible
+				perReplaceWithSz.Max = uint64(math.Ceil(float64(sz.Max) / float64(toReplaceSz.Min)))
```

Comment:
doesn't the max depend on whether replaceWithSz.Max is greater than toReplaceSz.Max or not?

if `replaceWithSz.Max > toReplaceSz.Max` (e.g. replace "a" with "aa"), the max result size occurs if we replace as many times as possible (`ceil(sz.Max/toReplaceSz.Min)`, as you have here)

if `replaceWithSz.Max <= toReplaceSz.Max` (e.g. replace "aa" with "a"), the max result size occurs if we replace zero times, right? Shouldn't that be represented here somewhere, and just use sz.Max as the result max?